proto/spire-next/types/status.proto

```diff
@@ -9,3 +9,19 @@ message Status {
     // A developer-facing error message.
     string message = 2;
 }
+
+message PermissionDeniedDetails {
+    enum Reason {
```

Comment:
food for thought: will we ever want an `AGENT_ALREADY_ATTESTED` reason that gets returned from node attestor plugins? if so, would anything need to change in this PR to support that? (beyond adding another enum)

Reply:
That's actually a great use case. I think this change is sufficient to handle that (by extending the enum).